ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,62 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+struct Mutating{F}
+    f::F
+end
+function (f::Mutating)(y, xs...)
+    y .= f.f(xs...)
+    return y
+end
+
+AD.@primitive function value_and_pullback_function(b::AD.EnzymeReverseBackend, f, xs...)
+    y = f(xs...)
+    return y,
+    Δ -> begin
+        Δ_xs = zero.(xs)
+        dup = if y isa Real
+            if Δ isa Real
+                Enzyme.Duplicated([y], [Δ])
```

Comment:
This seems a bit strange - that's not something an Enzyme user would do AFAIK.

Reply:
ya it's a quick and dirty hack to get it running, needs to be optimised

Reply:
Maybe one can reuse some of the things I did in https://github.com/TuringLang/DistributionsAD.jl/pull/254.

Reply:
If this is a real or tuple of real, this should be an active argument [in reverse mode]